lib/AST/SubstitutionMap.cpp

```diff
@@ -395,6 +398,40 @@ SubstitutionMap::combineSubstitutionMaps(const SubstitutionMap &firstSubMap,
     });
 }
 
+void SubstitutionMap::verify() const {
+#ifndef NDEBUG
```

Comment:
This seems like a good opportunity to introduce an -enable-verifiers flag so that they're runtime-enabled rather than compiled out of release builds.